rust/xaynet-client/src/api/http.rs

```diff
@@ -21,13 +23,65 @@ pub struct HttpApiClient {
 }
 
 impl HttpApiClient {
-    pub fn new<S>(address: S) -> Self
+    /// Creates a new HTTP(S) client.
+    ///
+    /// # Warning
+    /// If no trusted root `certificate`s are provided, then every server will be trusted.
+    pub fn new<S>(
+        address: S,
+        certificates: Option<Vec<Certificate>>,
+    ) -> Result<Self, HttpApiClientError>
     where
         S: Into<String>,
     {
-        Self {
-            client: Client::new(),
+        let client = if let Some(certificates) = certificates {
+            let mut builder = ClientBuilder::new().use_rustls_tls();
+            for certificate in certificates {
+                builder = builder.add_root_certificate(certificate);
+            }
+            builder.build().map_err(HttpApiClientError::Http)?
+        } else {
+            ClientBuilder::new()
+                .use_rustls_tls()
+                .danger_accept_invalid_certs(true)
+                .build()
+                .map_err(HttpApiClientError::Http)?
+        };
+
+        Ok(Self {
+            client,
             address: address.into(),
+        })
+    }
+
+    /// Reads DER and PEM certificates from given paths.
+    pub fn certificates_from(
+        paths: &Option<Vec<PathBuf>>,
+    ) -> Result<Option<Vec<Certificate>>, HttpApiClientError> {
+        if let Some(paths) = paths {
+            if paths.is_empty() {
+                Ok(None)
+            } else {
+                paths
+                    .iter()
+                    .map(|path| -> Result<Certificate, HttpApiClientError> {
```

Comment:
I think it's worth moving this closure into a free standing function. `load_cert()` or something like this.

Reply:
agree.